lib/pluginmanager/utils/http_client.rb

```diff
@@ -6,11 +6,12 @@ class RedirectionLimit < RuntimeError; end
     HTTPS_SCHEME = "https"
     REDIRECTION_LIMIT = 5
 
-    # Proxies should be handled by the library
-    # https://ruby-doc.org/stdlib-2.3.1/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-Proxies
     def self.start(uri)
       uri = URI(uri)
-      Net::HTTP.start(uri.host, uri.port, http_options(uri)) { |http| yield http }
+      proxy_url = ENV["https_proxy"] || ENV["HTTPS_PROXY"] || ""
+      proxy_uri = URI(proxy_url)
```

Comment:
This could yield a parsing error, we should do the following:

- catch the error
- Print a message telling the proxy isn't the right format.
- exit the process
- We might need to validate the scheme, to make sure it's http(s)

Reply:
if the proxy is not well formed, it will never reach this code:

```
logstash-5.6.1$ https_proxy="http:ohndoe:teste@192.168.66.3:3128" bin/logstash-plugin install logstash-filter-elapsed
java/util/Hashtable.java:459:in `put': java.lang.NullPointerException
	from java/util/Properties.java:166:in `setProperty'
	from java/lang/System.java:796:in `setProperty'
	from java/lang/reflect/Method.java:498:in `invoke'
	from /home/osboxes/logstash-5.6.1/lib/pluginmanager/proxy_support.rb:40:in `apply_env_proxy_settings'
	from /home/osboxes/logstash-5.6.1/lib/pluginmanager/proxy_support.rb:79:in `configure_proxy'
	from /home/osboxes/logstash-5.6.1/lib/pluginmanager/main.rb:26:in `(root)
```

Also, the https_proxy doesn't have to be a http url, it seems to work fine in my tests with `https_proxy="http://johndoe:teste@192.168.66.3:3128" `